### DISCUSSION TITLE
[서인] Web Component 회고록

### DISCUSSION THREAD

**Discussion by Seoin02** [React-Core-Learn/Web-Component#7] [회고록]
## 느낀 점
오랜만에 바닐라 자바스크립트로 구현해보니 낯설게 느껴졌습니다. 심지어 클래스로 구현해보는 게 처음이라 어려웠습니다.
또, 스터디 메이트는 양질의 코드 리뷰를 해주는데, 나는 그러지 못해 미안하기도 했습니다.
타입스크립트를 제대로 공부해야겠다고 생각했습니다.

## 배운 점
- 이벤트 처리 시 각각 하위요소가 아니라 상위요소에만 적용하면 버블링을 활용해 효율적으로 동작하게 할 수 있는 것을 직접 개선하면서 배웠습니다.
- class로 구현해보라는 얘길 많이 들었는데, 메서드가 직관적이라 구조를 파악하기에 좋은 것 같다고 생각했습니다. 하지만 은닉을 써보지 않은 게 아쉬웠습니다.
- state, setState, render를 바닐라 자바스크립트로 구현하면서 상태를 선언적으로 관리하는 게 좋다는 것을 직접 느꼈습니다.

**Comment by D5ng** [comment]
서로 시각을 넓혀가는게 이 스터디의 장점이라고 생각해요! 서인님꺼 보면서 렌더링에 대한 생각을 못해봤는데, 그런 부분에서 많은 도움이 되었습니다. 사고확장이 되었습니다.!

서로 다음 상태관리도 같이 파이팅해봐요!!
